python_ta/checkers/infinite_loop_checker.py

```diff
@@ -45,6 +50,162 @@ def _check_condition_all_var_used(self, node: nodes.While) -> None:
                 node=node.test,
             )
 
+    def _check_condition_constant(self, node: nodes.While) -> None:
+        """Helper function that checks if a constant while-loop condition may lead to an infinite loop.
+
+        This helper flags loops that meet **both** of the following criteria:
+            - The `while` condition is constant (e.g., `while 1 < 2`)
+            - The loop body contains no `return`, `break`, `raise`, `yield`, or `sys.exit()` calls
+        """
+        if not self._check_constant_loop_cond(
+            node, node.test
+        ) and not self._check_constant_form_condition(node):
+            return
+        inferred = utils.safe_infer(node.test)
+        if isinstance(inferred, util.UninferableBase) or inferred is None:
+            return
+        if (
+            isinstance(inferred, nodes.Const)
+            and bool(inferred.value) is False
+            or isinstance(inferred, (nodes.List, nodes.Tuple, nodes.Set))
+            and not inferred.elts
+            or isinstance(inferred, nodes.Dict)
+            and not inferred.items
+        ):
+            return
+
+        check_nodes = (nodes.Break, nodes.Return, nodes.Raise, nodes.Yield)
+        for child in node.body:
+            for exit_node in child.nodes_of_class(klass=(nodes.Call, *check_nodes)):
+                if isinstance(exit_node, check_nodes):
+                    return
+                # Check Call node to see if `sys.exit()` is called
+                elif (
+                    isinstance(exit_node, nodes.Call)
+                    and isinstance(exit_node.func, nodes.Attribute)
+                    and exit_node.func.attrname == "exit"
+                ):
+                    inferred = utils.safe_infer(exit_node.func.expr)
+                    if (
+                        not isinstance(inferred, util.UninferableBase)
+                        and inferred is not None
+                        and isinstance(inferred, nodes.Module)
+                        and inferred.name == "sys"
+                    ):
+                        return
+        else:
+            self.add_message("infinite-loop", node=node.test, confidence=INFERENCE)
+
+    def _check_constant_form_condition(self, node: nodes.While) -> bool:
+        """Helper function that checks if while loop condition is of constant form (e.g.: `1 < 2`, `5 - 1 >= 2 + 2`)"""
+        cond_vars = set(child for child in node.test.nodes_of_class(nodes.Name))
```

Comment:
Since you just need to know whether the collection is empty or not, you can directly call `any` on `node.test.nodes_of_class(nodes.Name)`. No need for the comprehension expression (`child for child in ...`); this is redundant.